src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -1124,6 +1297,192 @@ protected Name readSystemPropertyName() {
+        nextChar();
+        while (bp < buflen && ch == c) {
+            n++;
+            nextChar();
+        }
+        return n;
+    }
+
+    void skipLine() {
+        while (bp < buflen) {
+            if (ch == '\n' || ch == '\r') {
+                return;
+            }
+            nextChar();
+        }
+    }
+
+    int skipMarkdownCode(char term, int count, LineKind initialLineKind) {
+        LineKind lineKind = null;
+        while (bp < buflen) {
+            switch (ch) {
+                case '\n', '\r' -> {
+                    nextChar();
+                    int indent = readIndent();
+                    lineKind = (ch == '\n' || ch == '\r') ? LineKind.BLANK
+                            : (indent <= 3) ? peekLineKind()
+                            : LineKind.OTHER;
+                    switch (initialLineKind) {
+                        case CODE_FENCE -> {
+                            if (lineKind == LineKind.CODE_FENCE && ch == term && count(ch) >= count) {
+                                return bp;
+                            }
+                        }
+
+                        case OTHER -> {
+                            if (lineKind != LineKind.OTHER) {
+                                return -1;
+                            }
+                        }
+
+                        default -> {
+                            return -1;
+                        }
+
+                    }
+
+                }
+
+                default -> {
+                    if (ch == term && initialLineKind != LineKind.CODE_FENCE ) {
+                        if (count(ch) == count) {
+                            return bp;
+                        }
+                    }
+                    nextChar();
+                }
+
+            }
+        }
+        // found end of input
+        return -1;
+    }
+
+    /**
+     * The kinds of line, used when reading a Markdown documentation comment.
+     * The values are loosely ordered by a combination of the specificity of
+     * the associated pattern, and the likelihood of use in a doc comment.
+     *
+     * Note that some of the patterns overlap, such as SETEXT_UNDERLINE and
+     * THEMATIC_BREAK. The CommonMark spec resolves the ambiguities with
+     * priority rules, such as the following:
+     *
+     * * If a line of dashes that meets the above conditions for being a
+     *   thematic break could also be interpreted as the underline of a
+     *   setext heading, the interpretation as a setext heading takes precedence.
+     *
+     * * When both a thematic break and a list item are possible interpretations
+     *   of a line, the thematic break takes precedence.
+     *
+     * Here in this context, the primary purpose of these kinds is to help
+     * determine when a block boundary terminates literal text, such as a
+     * code span, and so the exact nature of the line kind does not matter.
+     */
+    enum LineKind {
+        BLANK(Pattern.compile("[ \t]*")),
```

Comment:
`BLANK` is a pseudo kind, because it is set manually, but never returned from `peekLine()`. I wonder if we can change it somehow.

Reply:
Even if it is set manually, it is still convenient to have it as a member in the `LineKind` enum class.

Reply:
Not that it invalidates your reply; clarification: I meant `peekLineKind()`, not `peekLine()`.